src/authorizer.js

```diff
@@ -41,6 +42,60 @@ class Authorizer {
     throw new Error('Unauthorized');
   }
 
+  async getApiKey(apiGateway, clientId) {
+    try {
+      const apiKeys = await apiGateway.getApiKeys({ nameQuery: clientId, includeValues: true, limit: 1 }).promise();
+      const apiKey = apiKeys.items[0];
+
+      if (apiKey && apiKey.id) {
+        return apiKey;
+      }
+      console.log(apiKeys);
+      throw new Error('Usage Api Key Id is not present');
+    } catch (e) {
+      console.error(e.message);
+      console.info('creating key');
+    }
+
+    return apiGateway.createApiKey({
+      description: `Key for client ${clientId}`,
+      enabled: true,
+      generateDistinctId: true,
+      name: clientId,
+      value: clientId
+    }).promise();
+  }
+
+  async ensurePlanKey(apiGateway, apiKey) {
+    try {
+      const usagePlanKey = await apiGateway.getUsagePlanKey({
+        keyId: apiKey.id,
+        usagePlanId: this.configuration.usagePlan
+      }).promise();
+
+      if (usagePlanKey && usagePlanKey.id) {
+        return usagePlanKey;
+      }
+      throw new Error('Usage Plan Key Id is not present');
+    } catch (e) {
+      console.error(e.message);
```

Comment:
These will end up in CW, we should use the `requestLogger` to log them.

Reply:
Actually your throw will just get caught in the next line, we probably don't want the `type/catch`